intermediate-python-testing.md

```diff
@@ -361,3 +362,74 @@ Then, you need to patch the class you'd like to mock, and set its `return_value`
 mock_handle = mock.patch('path.to.CLASS_YOU_ARE_MOCKING')
 mock_handle.return_value = mocked_class
 ```
+
+## requests-mock
+
+Sometimes, your app sends requests to external URIs. And sometimes, these requests either have very little to do with the needs of your tests or fully interfere with them. [`requests-mock`](https://requests-mock.readthedocs.io/en/latest/overview.html) leaps into action!
+
+`requests-mock` sets up [a transport adapter](http://docs.python-requests.org/en/latest/user/advanced/#transport-adapters) (a mechanism which defines how to handle HTTP requests). With a `requests-mock` adapter, you can load data into responses from specified URIs. How to use it? `requests-mock` comes with [multiple patterns for instantiating the `requests_mock.Mocker` class](https://requests-mock.readthedocs.io/en/latest/mocker.html#activation). For instance, use it as a context manager:
+
+```
+import requests
+import requests_mock
+
+def test_get_resp():
+    with requests_mock.Mocker() as m:
+        m.get("http://propane.org", text="that boy ain't right")
+        rsp = requests.get("http://propane.org").text
+
+        assert rsp.text == "that boy ain't right"
+``` 
+
+Your app might hit variations of the same URI, however. `requests-mock` saves you the tedium of mocking each path. You can define an adaptor that matches multiple paths, [e.g., with Regex](https://requests-mock.readthedocs.io/en/latest/matching.html#regular-expressions). Let's return to the example above, but build out some of its functionality.
+
+**`utils.py`**
+```python
+def get_result():
+    # query the database and return result
+
+def scrape_an_endpoint(endpoint):
```

Comment:
i'm not sure i follow the arg being `endpoint`. based on the code in `tasks.py`, it looks like it should be `result_id` or something similar?

Reply:
yup, that's much more clear – variable names, important in the real world, important in the abstract